### PR TITLE
[http] support SameSite in Cookies.

### PR DESCRIPTION
**Pull request by sun-jiao**
Add support for SameSite attribute in `Cookies`.

### CODE REVIEW

File: sdk/lib/_http/http.dart [hunk 1]
```diff
@@ -735,6 +735,19 @@ abstract class ContentType implements HeaderValue {
   String? get charset;
 }
 
+/// Type of [Cookie.sameSite], which defines whether the cookie is available
```
Comment (lrhn) [inline sdk/lib/_http/http.dart:738]: "Type" -> "Value"

(Class documentation should describe instances, not the class itself.)

**Review by lrhn — COMMENTED** [review]
Entirely stylistic comments.
No opinion on whether we want this (potentially breaking) change or not.

**Comment by sun-jiao** [comment]
> 

Hi, I am still working. Please feel free to add any comments.

**Comment by mit-mit** [comment]
Thanks. I see the comments in were addressed; thanks for that.

I also saw https://github.com/dart-lang/sdk/issues/51486, which we'll see if we can get approved.

Did you address the comments in https://github.com/dart-lang/sdk/pull/51457#pullrequestreview-1344687580? If so, you can go ahead and resolve them.

**Comment by mit-mit** [comment]
@sun-jiao thanks for the contribution, and thanks for sticking through with that very long review!

**Comment by sun-jiao** [comment]
You're welcome. @mit-mit  I'm glad to contribute to dart project and also look forward to contributing more in the future.

**Pull request closed without merging**